src/secure_random.cr

```diff
@@ -58,14 +74,80 @@ module SecureRandom
       raise ArgumentError.new "negative size: #{n}"
     end
 
-    slice = Slice(UInt8).new(n)
-    result = LibCrypto.rand_bytes slice, n
-    if result != 1
-      error = LibCrypto.err_get_error
-      error_string = String.new LibCrypto.err_error_string(error, nil)
-      raise error_string
+    init unless @@initialized
+
+    ifdef linux
+      if @@getrandom_available
+        return getrandom(n)
+      end
+    end
+
+    buf = Slice(UInt8).new(n)
+
+    if urandom = @@urandom
+      urandom.read_fully(buf)
+      return buf
+    end
+
+    ifdef !without_openssl
```

Comment:
I checked and we don't use the `flag?` macro method but `ifdef` everywhere in the core/stdlib.